lib/philomena/images/query.ex

```diff
@@ -12,6 +13,35 @@ defmodule Philomena.Images.Query do
     end
   end
 
+  defp filter_id_transform(%{filter: true}, _value),
+    do: {:error, "Filter queries inside filters are not allowed."}
+
+  defp filter_id_transform(%{user: user} = ctx, value) do
+    case Integer.parse(value) do
+      {value, ""} when value >= 0 ->
+        filter = Repo.get(Filter, value)
```

Comment:
This makes 1 query for every filter_id present in the query. This could result in quite a lot of queries. This amplification would be avoided if the query was once for all filter_ids present in the query.

Reply:
(If you're wondering how you might efficiently know which filters will be referenced inside the query, one strategy is to [guess](https://github.com/philomena-dev/philomena/blob/master/lib/philomena_web/markdown_renderer.ex#L42).)